src/sidebar/components/filter-status.js

```diff
@@ -8,176 +9,278 @@ import useRootThread from './hooks/use-root-thread';
+              by <span className="filter-facet">{focusDisplayName}</span>
             </span>
           )}
-          {filterState.forcedVisibleCount > 0 && (
+          {additionalCount > 0 && (
             <span className="filter-facet--muted">
               {' '}
-              (and {filterState.forcedVisibleCount} more)
+              (and {additionalCount} more)
             </span>
           )}
         </div>
-        <div>
-          <Button className="button--primary" {...buttonProps} />
-        </div>
+        <div>{actionButton}</div>
       </div>
     </div>
   );
 }
 
+FilterStatusPanel.propTypes = {
+  resultCount: propTypes.number.isRequired,
+  actionButton: propTypes.object.isRequired,
+  additionalCount: propTypes.number,
+  entitySingular: propTypes.string,
+  entityPlural: propTypes.string,
+  filterQuery: propTypes.string,
+  focusDisplayName: propTypes.string,
+};
+
+/**
+ * This status is used when there are selected annotations (including direct-
+ * linked annotations).
+ *
+ * Message formatting:
+ * "[Showing] (No|<resultCount>) annotation[s] [\(and <additionalCount> more\)]"
+ * Button:
+ * "<cancel icon> Show all [\(<totalCount)\)]" - clears the selection
+ *
+ * @param {FilterModeProps} props
+ */
+function SelectionFilterStatus({ filterState, rootThread }) {
+  const clearSelection = useStore(store => store.clearSelection);
+  // The total number of top-level annotations (visible or not)
+  const totalCount = useStore(store => store.annotationCount());
+  // Count the number of visible annotations—top-level only
+  const visibleAnnotationCount = (rootThread.children || []).filter(
+    thread => thread.annotation && thread.visible
+  ).length;
+
+  // The number displayed in "(and x more)" is the difference between
+  // all visible top-level annotations and the count of selected annotations
+  // (i.e. additionalCount accounts for any visible top-level annotations
+  // that are not in the current selection)
+  const additionalCount = visibleAnnotationCount - filterState.selectedCount;
+
+  // Because of the confusion between counts of entities between selected
+  // annotations and filtered annotations, don't display the total number
+  // when in user-focus mode because the numbers won't appear to make sense.
+  const buttonText = filterState.focusConfigured
+    ? 'Show all'
+    : `Show all (${totalCount})`;
+
+  const button = (
+    <Button
+      buttonText={buttonText}
+      className="button--primary"
+      onClick={() => clearSelection()}
+      icon="cancel"
+    />
+  );
+  return (
+    <FilterStatusPanel
+      resultCount={filterState.selectedCount}
+      additionalCount={additionalCount}
+      actionButton={button}
+    />
+  );
+}
+
+SelectionFilterStatus.propTypes = {
+  filterState: propTypes.object.isRequired,
+  rootThread: propTypes.object.isRequired,
+};
+
+/**
+ * Status when there is an applied filter query. There may also be an applied
+ * user-focus filter.
```

Comment:
Won't this render in place of the focus filter? In other words, they don't both render at the same time right? I'm a bit confused by the comment.

Reply:
Hmmm. I am working on updating the comments in this module for clarity, but it's tricky.

Yes, a maximum of one status component is used at any time. This one (`QueryFilterStatus`) is used if there is a filter query present, and there are no selected annotations. There may or may not be an applied user filter at the same time.

`FocusFilterStatus` is only used if user-focus is configured, and _nothing else_.

Yeah, I'm pretty much just re-typing the comment. Can you help me understand your points of confusion so I can make it clearer?